src/posix/platform/misc.cpp

```diff
@@ -76,6 +76,12 @@ otError otPlatSetMcuPowerState(otInstance *aInstance, otPlatMcuPowerState aState
     return error;
 }
 
+void otPlatAssertFail(const char *aFilename, int aLineNumber)
+{
+    otLogCritPlat("assert failed at %s:%d", aFilename, aLineNumber);
+    exit(1);
```

Comment:
`exit(1)` won't dump core. Can we use `assert(false)` here?

Reply:
@simonlingoogle I believe this PR is to enable assertion in release build (assertion is disabled by `NDEBUG`), so `assert(false)` is not going to work.

Reply:
Use `abort()` which will generate a core dump instead. Thanks for pointing out.

Reply:
We can use `assert(false)` before `exit(1)`. 
In release build with `NDEBUG`, `assert(false)` will be ignored and exit(1) is effective. 
In debug build, `assert(false)` generates the core dump. 

Actually I am not quite clear about the intention of this change. In what case will this `otPlatAssertFail` be better than the default `assert` behavior?

Reply:
Done